### DISCUSSION TITLE
depecatedModule props

### DISCUSSION THREAD

**Discussion by aweell** [Telefonica/mistica-design#505] [Proposals]
### Descripción

Para los casos en los que el `deprecatedModule` (Mística Resources Library) tiene limitaciones de altura, sería interesante poder ocultar la caja de "reasons why" mostrando únicamente el texto que indica al usuario cuál es el reemplazo que debe de hacer.

Componente:

<img width="525" alt="Screenshot 2022-06-24 at 18 38 35" src="https://user-images.githubusercontent.com/44420072/175595592-4b91c335-c520-44de-b6dd-bd44a5c749bd.png">

Ejemplo con restricción de altura:

<img width="682" alt="Screenshot 2022-06-24 at 18 44 03" src="https://user-images.githubusercontent.com/44420072/175604079-7f9767f8-48c7-49a2-80a7-6a6ec802a6c9.png">

También adaptar el texto de su título para que no únicamente indique que es un componente deprecado sino que pueda hacer referencia a una variante.

### Propuesta de props

| Prop name | Prop type | Values | Description |
| :--------- | :----- | :--------- | :------ |
| Type       | `Variant` | `Component`, `Variant` | Elige el tipo de elemento que se depreca |
| Reason | `Boolean` | `True`, `False` | Muestra u oculta la caja de razones |

**Comment by yceballost** [comment]
El componente debería estar preparado para que ocultando la capa de _Reason why_ obtengas lo que dices, no obstante me parece buena idea manejar eso con las props de Figma para facilitar su utilización!

La única parte que me cuesta visualizar es la que comentas respecto a hacer referencia a una variante, cómo te la imaginas?

**Reply by aweell** [reply]
### Ocultar reasons why

> El componente debería estar preparado para que ocultando la capa de Reason why obtengas lo que dices, no obstante me parece buena idea manejar eso con las props de Figma para facilitar su utilización!

Cierto, justo me refería a añadirlo como una layer component prop

### Variant vs Component prop

<img width="787" alt="Screenshot 2022-06-24 at 21 05 02" src="https://user-images.githubusercontent.com/44420072/175649338-14f09efe-9e0c-4c72-a8f3-421a4284a11b.png">

Aquí el objetivo es discernir a nivel visual entre aquellos componentes que:

* Tienes que reemplazar completamente (elegir todas las instancias de un componente y reemplazarlo por uno nuevo) 
* Se ha realizado un refactor y la configuración de props ha cambiado (tienes que cambiar la variante actual por otra dentro del mismo componente)

**Reply by yceballost** [reply]
mola! (e igual aprovecharía para mejorar el diseño de ese modulito que se hizo en su día muy rápido 🤣)
